lib/modules/datasource/rubygems/get.ts

```diff
@@ -182,4 +218,110 @@ export class InternalRubyGemsDatasource extends Datasource {
 
     return new Marshal(response.body).parsed as T;
   }
+
+  private nexusEndPointRepoFrom(url: string): {
+    nexusRegistry?: string;
+    repository?: string;
+  } {
+    const endPointRe = regEx(
+      `^(?<nexusRegistry>.*:\\d{1,5}).*\\/repository\\/(?<repository>\\w+)\\/?`
+    );
+    const { nexusRegistry, repository } = url.match(endPointRe)?.groups ?? {};
+    return { nexusRegistry, repository };
+  }
+
+  @cache({
+    namespace: 'datasource-nexus',
+    key: (nexusEndPoint: string) => `isNexusDataSource:${nexusEndPoint}`,
+    ttlMinutes: 1440,
+  })
+  async isNexusDataSource(nexusEndPoint: string): Promise<boolean> {
+    const statusEndPoint = '/service/rest/v1/status';
+    const nexusUrl = joinUrlParts(nexusEndPoint, statusEndPoint);
+    try {
+      const response = await this.http.getJson<HttpResponse>(nexusUrl);
+      return this.extractServerFrom(response).startsWith('Nexus/') ?? false;
+    } catch (err) {
+      // istanbul ignore if
+      if (err.response?.headers?.server?.startsWith('Nexus/')) {
+        this.handleGenericErrors(err);
+      }
+    }
+    return false;
+  }
+
+  private extractServerFrom(
+    response: HttpResponse<HttpResponse<string>>
+  ): string {
+    let headerRespose = '';
+    // istanbul ignore else: not testable with nock
```

Comment:
why it's not testable? i think we should move those little helper functions to ./common.ts or ./nexus.ts for better testability and maintaining